src/main/scala/terminal/FileSystem.scala

```diff
@@ -0,0 +1,113 @@
+package terminal
+
+import java.util.NoSuchElementException
+
+
+class FileSystem {
+  //key = command;  value = how many paths need to be specified
+  val commandsAndNoOfPaths = Map("pwd" -> 0, "mkdir" -> 1, "ls" -> 0, "h" -> 0, "q" -> 0 )
+
+  var rootPath : Folders = _
```

Comment:
Could we change the logic behind this class so it will not use mutable variables?

Reply:
i.e. we could have one instance of some kind of _terminal_, for which the rootPath/cmdPrefix would be immutable.